lib/matplotlib/afm.py

```diff
@@ -370,40 +351,19 @@ def _parse_optional(fh):
     return d[b'StartKernData'], d[b'StartComposites']
 
 
-def _parse_afm(fh):
-    """
-    Parse the Adobe Font Metrics file in file handle *fh*.
-
-    Returns
-    -------
-    header : dict
-        A header dict. See :func:`_parse_header`.
-    cmetrics_by_ascii : dict
-        From :func:`_parse_char_metrics`.
-    cmetrics_by_name : dict
-        From :func:`_parse_char_metrics`.
-    kernpairs : dict
-        From :func:`_parse_kern_pairs`.
-    composites : dict
-        From :func:`_parse_composites`
-
-    """
-    _sanity_check(fh)
```

Comment:
I'm not sure dropping this `_sanity_check` is quite equivalent; `_parse_header` is pretty lenient, so it seems like it could run through the whole file, whereas this check would have given up much much earlier.

Reply:
That's a good point. Can we push the check down into `_parse_header` and raise from there?

Reply:
But is it really a problem if we raise after parsing the whole file rather than immediately?  I'm reasonably confident parsing malformed afm files is not a bottleneck for matplotlib's performance.

Reply:
It's logically cleaner to raise a file format error from `_parse_header()`. This is a low-level check which shouldn't interrupt the high-level parse-this, parse-that logic in `__init__`.

Reply:
OK, I inlined the check into _parse_header.  This still avoids having to do the dance with fh.tell()/fh.seek() that an independent _sanity_check requires.